### PR TITLE
Issue #1340 Make Splashpage components selected by default

### PR DESCRIPTION
**Pull request by nikhilgupta1211**
#1340 
Default state of components of splashpage changed to checked .

### CODE REVIEW

**Comment by nikhilgupta1211** [comment]
@Ana06 according to the test some components are only selected but as we are making all the components selected by default the test `./spec/features/versions_spec.rb:218` is failing . Should i modify the test ?

**Comment by Ana06** [comment]
@nikhilgupta1211 sure!

[In the tests](https://github.com/openSUSE/osem/blob/master/spec/features/versions_spec.rb#L224) we select 3 options. You should change it to unselect the others so the test keep testing the same thing. :wink:

**Comment by nikhilgupta1211** [comment]
Thanks :) I have implemented the changes .

**Comment by nikhilgupta1211** [comment]
@Ana06  i think there is problem in this that as we are using a form partial for both edit and new action, so when a user create a splashpage with some unchecked options they will appear to be checked in the edit view and the user have to uncheck them again . So should i try to fix this or as there will not be much unchecked options as mentioned in the issue we can use this 😉?

**Comment by Ana06** [comment]
@nikhilgupta1211 

> @Ana06 i think there is problem in this that as we are using a form partial for both edit and new action, so when a user create a splashpage with some unchecked options they will appear to be checked in the edit view and the user have to uncheck them again . So should i try to fix this or as there will not be much unchecked options as mentioned in the issue we can use this :wink:?

It should only by checked by default when creating it, when editing it should show the saved ones.

**Comment by nikhilgupta1211** [comment]
@Ana06 can you please tell me that should i implement it using javascript or ruby?

**Comment by nikhilgupta1211** [comment]
We can add a check-all option to check and uncheck all the checkboxes

**Comment by Ana06** [comment]
@nikhilgupta1211 

> @Ana06 can you please tell me that should i implement it using javascript or ruby?

Ruby!

**Comment by nikhilgupta1211** [comment]
@Ana06 i have made the changes :)

File: spec/features/versions_spec.rb [hunk 1]
```diff
@@ -218,6 +218,15 @@
   scenario 'display changes in splashpages', feature: true, versioning: true, js: true do
     visit admin_conference_splashpage_path(conference.short_title)
     click_link 'Create Splashpage'
+    uncheck('Display tracks on the splashpage?')
```
Comment (sayalilunkad) [inline spec/features/versions_spec.rb]: I think it would look better if we leave the 'on the splashpage?' bit. It seems too repetitive. @Ana06 What do you think?

**Comment by nikhilgupta1211** [comment]
I have made the changes

**Comment by Ana06** [comment]
@nikhilgupta1211 please read the [commits](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md#commits) and [pull request](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md#pull-requests-workflow) sections that we recently added to our [contributing guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md). :wink:

**Comment by nikhilgupta1211** [comment]
@Ana06 should i squash the commits ?

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,4 +1,11 @@
 module ApplicationHelper
+  # Set the value of checkbox to checked for the splashpages#new action
+  def chk
```
Comment (differentreality) [inline app/helpers/application_helper.rb]: What does chk mean ?!

Comment (nikhilgupta1211) [inline app/helpers/application_helper.rb]: sorry I forgot to change the name of the method

**Comment by nikhilgupta1211** [comment]
I have also added the Venue and Lodgings to the list 
image

File: app/views/admin/splashpages/_form.html.haml [hunk 3]
```diff
@@ -6,15 +6,15 @@
...
-        = f.input :include_tracks, label: 'Display tracks on the splashpage?'
-        = f.input :include_program, label: 'Display program on the splashpage?'
-        = f.input :include_cfp, label: 'Display call for papers information on splashpage, while cfp is open?'
-        = f.input :include_venue, label: 'Display venue on the splashpage?'
-        = f.input :include_registrations, label: 'Display the registration period on the splashpage?'
-        = f.input :include_tickets, label: 'Display tickets on the splashpage?'
-        = f.input :include_lodgings, label: 'Display the lodgings on the splashpage?'
-        = f.input :include_sponsors, label: 'Display sponsors on the splashpage?'
-        = f.input :include_social_media, label: 'Display social media on the splashpage?'
+        = f.input :include_tracks, label: 'Display tracks', input_html: { checked: checked }
```
Comment (differentreality) [inline app/views/admin/splashpages/_form.html.haml]: What if we just have
`input_html: { params[:action] == 'new' ? 'checked' : false }`

Comment (nikhilgupta1211) [inline app/views/admin/splashpages/_form.html.haml]: done :)

**Comment by differentreality** [comment]
:+1: Please squash and rebase!

**Comment by nikhilgupta1211** [comment]
Done :)

File: app/views/admin/splashpages/_form.html.haml [hunk 4]
```diff
@@ -6,15 +6,15 @@
...
-        = f.input :include_tracks, label: 'Display tracks on the splashpage?'
-        = f.input :include_program, label: 'Display program on the splashpage?'
-        = f.input :include_cfp, label: 'Display call for papers information on splashpage, while cfp is open?'
-        = f.input :include_venue, label: 'Display venue on the splashpage?'
-        = f.input :include_registrations, label: 'Display the registration period on the splashpage?'
-        = f.input :include_tickets, label: 'Display tickets on the splashpage?'
-        = f.input :include_lodgings, label: 'Display the lodgings on the splashpage?'
-        = f.input :include_sponsors, label: 'Display sponsors on the splashpage?'
-        = f.input :include_social_media, label: 'Display social media on the splashpage?'
+        = f.input :include_tracks, label: 'Display tracks', input_html: { checked: params[:action] == 'new' ? 'checked' : false }
```
Comment (Ana06) [inline app/views/admin/splashpages/_form.html.haml]: You can not set it to false, you have to take the value from conference splashpage:

`@conference.splashpage.try(:include_tracks)` returns `false` or `true` if it is set, or `nil` in case that it is not

File: app/views/admin/splashpages/show.html.haml [hunk 5]
```diff
@@ -60,6 +60,20 @@
           - else
             No
         %dt
+          Include Venue:
```
Comment (Ana06) [inline app/views/admin/splashpages/show.html.haml]: Those are things that where missed in the show page, right?

They should be in a different commit as they are unrelated to making Splashpage components selected by default :wink:

File: spec/features/versions_spec.rb [hunk 6]
```diff
@@ -218,12 +218,21 @@
   scenario 'display changes in splashpages', feature: true, versioning: true, js: true do
     visit admin_conference_splashpage_path(conference.short_title)
     click_link 'Create Splashpage'
+    uncheck('Display tracks')
```
Comment (Ana06) [inline spec/features/versions_spec.rb]: instead of unchecking everything when creating it, what about leaving by default here and after clicking the edit link unchecking everything except _Make splash page public_, _Display tracks on the splashpage?_ and _Display the registration period on the splashpage?_

File: app/views/admin/splashpages/show.html.haml (hunk 5 above)
Comment (nikhilgupta1211) [inline app/views/admin/splashpages/show.html.haml]: Yes, ok :)

File: spec/features/versions_spec.rb (hunk 6 above)
Comment (nikhilgupta1211) [inline spec/features/versions_spec.rb]: @Ana06 that does not pass the test I think for that we have to also update the expect(page).to

Comment (Ana06) [inline spec/features/versions_spec.rb]: why?

Comment (nikhilgupta1211) [inline spec/features/versions_spec.rb]: Currently when the elements were not selected by default we were updating the _Make splash page public, Display tracks on the splashpage? and Display the registration period on the splashpage?_  but now as all the components are selected by default so by unchecking them in the edit link we are updating all of them except for the _tracks and registration component_ so we have to change the version_spec.rb:237 expect(page).to

**Comment by nikhilgupta1211** [comment]
I have made the changes in the tests :)

File: app/views/admin/splashpages/_form.html.haml [hunk 7]
```diff
@@ -6,15 +6,15 @@
...
-        = f.input :include_tracks, label: 'Display tracks on the splashpage?'
-        = f.input :include_program, label: 'Display program on the splashpage?'
-        = f.input :include_cfp, label: 'Display call for papers information on splashpage, while cfp is open?'
-        = f.input :include_venue, label: 'Display venue on the splashpage?'
-        = f.input :include_registrations, label: 'Display the registration period on the splashpage?'
-        = f.input :include_tickets, label: 'Display tickets on the splashpage?'
-        = f.input :include_lodgings, label: 'Display the lodgings on the splashpage?'
-        = f.input :include_sponsors, label: 'Display sponsors on the splashpage?'
-        = f.input :include_social_media, label: 'Display social media on the splashpage?'
+        = f.input :include_tracks, label: 'Display tracks', input_html: { checked: params[:action] == 'new' ? 'checked' : @conference.splashpage.try(:include_tracks) }
```
Comment (Ana06) [inline app/views/admin/splashpages/_form.html.haml]: This you tried this? I would say it doesn't work :confounded: 

If `checked` attribute is present it will be checked even if it is set to false (`checked="false"`), so the value doesn't mind)

Comment (Ana06) [inline app/views/admin/splashpages/_form.html.haml]: why about writing this like:

``` Ruby
f.input :include_tracks, label: 'Display tracks', input_html: { checked: params[:action] == 'new' || @conference.splashpage.try(:include_tracks) }
```

As `true` and ''checked` both work :wink:

Comment (Ana06) [inline app/views/admin/splashpages/_form.html.haml]: And the same for the rest

Comment (nikhilgupta1211) [inline app/views/admin/splashpages/_form.html.haml]: @Ana06 I guess we don't need to do `@conference.splashpage.try(:include_tracks)` we can do `@splashpage.try(:include_tracks)`

**Comment by nikhilgupta1211** [comment]
I have made the changes :)
